overlord/devicestate/devicestate.go

```diff
@@ -202,11 +203,63 @@ func checkGadgetOrKernel(st *state.State, snapInfo, curInfo *snap.Info, _ snap.C
 	return nil
 }
 
+var (
+	// TODO: replace with gadget.IsCompatible() once we are ready
+	gadgetIsCompatible = func(current, new *gadget.Info) error { return nil }
+)
+
+func checkGadgetRemodelCompatible(st *state.State, snapInfo, curInfo *snap.Info, snapf snap.Container, flags snapstate.Flags, deviceCtx snapstate.DeviceContext) error {
```

Comment:
maybe this function should be in remodel.go?  not sure

Reply:
Eventually, not having a better place, I moved it around to handlers_remodel.go